src/application-base/ApplicationBase.jsx

```diff
@@ -17,10 +17,16 @@ import useTestOverrides from './private/useTestOverrides';
 
 import styles from './ApplicationBase.module.scss';
 
+/* global TERRA_THEME_CONFIG */
+
 const cx = classNames.bind(styles);
 
 const browserLocale = getBrowserLocale();
 
+// We only need to retrieve the root theme and root theme name once for the life of the application.
+const themeConfig = (typeof (TERRA_THEME_CONFIG) !== 'undefined') ? TERRA_THEME_CONFIG : undefined;
+const rootThemeName = themeConfig?.theme ? themeConfig.theme : 'terra-default-theme';
```

Comment:
This logic, to read the config and apply the theme name when the theme class name is undefined could be moved to terra-theme-context provider, but it felt weird to make these assumptions there. 

I tried to write a jest test for these, redefining the TERRA_THEME_CONFIG global, but i think this is untestable with these functions defined outside of the class. This is more efficient though because these methods are only run once.